### PR TITLE
enh(setting): Migrate OCM / OCS provider tests to SetupCheck

### PR DESCRIPTION
**Pull request by susnux**
* Part of #41364 

## Summary

Migrate the ocs-provider and ocm-provider from legacy to SetupCheck.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/tests/SetupChecks/OcxProvicersTest.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	/** @var IConfig|MockObject */
+	private $config;
+	/** @var IURLGenerator|MockObject */
+	private $urlGenerator;
+	/** @var IClientService|MockObject */
+	private $clientService;
+	/** @var LoggerInterface|MockObject */
+	private $logger;
+	/** @var OcxProviders|MockObject */
+	private $setupcheck;
```
Comment (come-nc) [inline apps/settings/tests/SetupChecks/OcxProvicersTest.php:51]:
```suggestion
	private IL10N $l10n;
	private IConfig $config;
	private IURLGenerator $urlGenerator;
	private IClientService $clientService;
	private LoggerInterface $logger;
	private OcxProviders $setupcheck;
```

File: apps/settings/lib/SetupChecks/OcxProviders.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		if (empty($missingProviders)) {
+			return SetupResult::success();
+		}
+
+		return SetupResult::warning(
+			$this->l10n->t('Your web server is not properly set up to resolve %1$s.
+This is most likely related to a web server configuration that was not updated to deliver this folder directly.
+Please compare your configuration against the shipped rewrite rules in ".htaccess" for Apache or the provided one in the documentation for Nginx.
+On Nginx those are typically the lines starting with "location ~" that need an update.', [join(', ', array_map(fn ($s) => '"'.$s.'"', $missingProviders))]),
+			'admin-nginx',
```
Comment (come-nc) [inline apps/settings/lib/SetupChecks/OcxProviders.php:96]: You need to use the url generator to build the link here, see the other checks for inspiration

**Review by come-nc — DISMISSED** [review]

File: apps/settings/tests/SetupChecks/OcxProvicersTest.php (hunk 1 above)
Comment (susnux) [reply apps/settings/tests/SetupChecks/OcxProvicersTest.php:51]: Not sure here we use the comments everywhere else and at least for me removing them breaks my IDE as then `$this->setupcheck->expects($this->exactly(2))` will be undefined as `OcxProviders` has no method `expects` but only `MockObject` has.

File: apps/settings/lib/SetupChecks/OcxProviders.php (hunk 2 above)
Comment (susnux) [reply apps/settings/lib/SetupChecks/OcxProviders.php:96]: Ah yes sorry!

File: apps/settings/tests/SetupChecks/OcxProvicersTest.php (hunk 1 above)
Comment (nickvergessen) [reply apps/settings/tests/SetupChecks/OcxProvicersTest.php:51]:
```suggestion
	private IL10N|MockObject $l10n;
	private IConfig|MockObject $config;
	private IURLGenerator|MockObject $urlGenerator;
	private IClientService|MockObject $clientService;
	private LoggerInterface|MockObject $logger;
	private OcxProviders|MockObject $setupcheck;
```

Comment (come-nc) [reply apps/settings/tests/SetupChecks/OcxProvicersTest.php:51]: Hum yeah never thought about that, I always typed as the target interface/class.
You’d need an intersection type to be precise I suppose, but that’s not possible on 8.0.

**Comment by susnux** [comment]
Resolved all comments, please re-review :)

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
```diff
- enh(setting): Migrate OCM / OCS provider tests to SetupCheck 
+ feat(setting): Migrate OCM / OCS provider tests to SetupCheck 
```

**Review by come-nc — APPROVED** [review]

**Pull request merged**